pkg/catalog/fake.go

```diff
@@ -82,6 +76,20 @@ func NewFakeMeshCatalog(kubeClient kubernetes.Interface) *MeshCatalog {
 
 		return vv
 	}).AnyTimes()
+
+	mockKubeController.EXPECT().ListPods().DoAndReturn(func() []*v1.Pod {
+		vv, err := kubeClient.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return nil
+		}
+
+		var podRet []*v1.Pod = []*v1.Pod{}
+		for idx := range vv.Items {
+			podRet = append(podRet, &vv.Items[idx])
+		}
+		return podRet
+	}).AnyTimes()
```

Comment:
Why is this needed?

Reply:
This is needed for the tests run catalog functions, as I can't interface with the kubecontroller inside it. I changed some tests to just run a separate kubecontroller as they just test xds_cert functions, and they don't need catalog.
But for the ones running catalog, they need this.

Reply:
I'd drop a comment here and file an issue to track such custom fake code needing enhancements.

Reply:
+1, could we maybe reuse https://github.com/openservicemesh/osm/issues/1683 and generalize the issue?